### DISCUSSION TITLE
Add option to copy a project

### DISCUSSION THREAD

**Discussion by Digi92** [mosparo/mosparo#108] [Ideas]
Hi,
i think a function to copy an existing project would be nice.
A user might want to copy an existing project with all settings, rules and rule sets to use this as a basis for a new project.

To consider would be of course how the rights should look like.

What is your opinion about this?

**Comment by zepich** [comment]
Hi @Digi92 

Thank you very much for your suggestion.

In general, it's a good idea. But copying as you suggested will help you only in one situation (when you create a new project).

I think it would be better to add an 'Export' and 'Import' functionality. With the export functionality, you can export the configuration of a project (settings, rules, and rule sets). With the import functionality, you can import the configuration you exported before.

With this, you can copy a project as you suggested - but you can also:

1. Copy projects from one mosparo installation to a different one (except, of course, the submissions, but they're lost anyways after 14 days)
2. Synchronize a project over and over again.
3. Export your configuration as a backup before you change anything (if we add a command, you could export the configuration automatically as a backup).

What do you think about that approach? It would make the copy process you suggested to create a project a bit _harder_ since you need 2 - 3 clicks more. But I think it's easier and can be used in more situations.

**Reply by Digi92** [reply]
I think you are right, copying would only be needed for the one case.
An import/export functionality would solve the problem much better.

In addition, the import/export function could also be made more flexible.
You could also extend the import/export in a project with checkboxes for the individual areas (rules, rulesets and settings), so that the user can flexibly decide what he wants to export/import.
Additionally, you could ask for the rules and rulesets if they should be added to or overwrite the existing ones.

I also think that many users inital would not come directly to the idea of using the import/export function, here it could be implemented a copy button, next to the project delete button, which links to the documentation.

**Reply by zepich** [reply]
Awesome! Thank you very much for your suggestions. Selecting the data you want to export/import and how they (mainly rules) should be imported (overwrite existing rule/append to rule/added as a new rule) is a great thing too.

We will add that!
